pkg/metrics/statsd/statsd.go

```diff
@@ -103,13 +106,65 @@ func (s *statsd) run() {
 	}
 }
 
-func (s *statsd) dial() (err error) {
-	path := "/var/etw/mdm_statsd.socket"
+func (s *statsd) parseSocketEnv(env string) (string, string, error) {
+	// Verify network:address format
+	parameters := strings.SplitN(env, ":", 2)
+	if len(parameters) != 2 {
+		return "", "", fmt.Errorf("malformed definition for the mdm statds socket. Expecting udp:<hostname>:<port> or unix:<path-to-socket> format. Got: %q", env)
+	}
+	network := strings.ToLower(parameters[0])
+	address := parameters[1]
+	return network, address, nil
+}
+
+func (s *statsd) validateSocketDefinition(network string, address string) (bool, error) {
+	//Verify supported protocol provided. TCP might just work as well, but this was never tested
+	if network != "udp" && network != "unix" {
+		return false, fmt.Errorf("unsupported protocol for the mdm statds socket. Expecting  'udp:' or 'unix:'. Got: %q", network)
+	}
+
+	return true, nil
+}
+
+func (s *statsd) getDefaultSocketValues() (string, string) {
+	network := "unix"
+	address := "/var/etw/mdm_statsd.socket"
+
 	if s.env.IsLocalDevelopmentMode() {
-		path = "mdm_statsd.socket"
+		address = "mdm_statsd.socket"
+	}
+
+	return network, address
+}
```

Comment:
A little nit here, couldn't this be a little simpler ? 
```go
if s.env.IsLocalDevelopmentMode() {
    return "unix", "mdm_statsd.socket"
}
return "unix", "/var/etw/mdm_statsd.socket"
``` for example ?

Reply:
It would be, but we don't have any hint in the code anymore what the function deals with (after removing the name from return parameter list. (We could add comments, but ...... -8-)